bpaf_core/src/split.rs

```diff
@@ -297,7 +305,34 @@ pub(crate) fn split_param<'a>(
     match value {
         OsOrStr::Str(cow) => split_str_param(cow.as_ref(), args, flags),
         OsOrStr::Os(cow) => match ascii_prefix(cow.as_ref()) {
-            Some(_) => todo!(),
+            // SAFETY: cow contains only utf8
+            Some((_, None)) => split_str_param(cow.to_str().unwrap(), args, flags),
+            Some((prefix, Some(suffix))) => match split_str_param(prefix.as_ref(), args, flags)? {
+                Arg::Named { name, value } => {
+                    // we are in the {name}=some-valid-utf8 situation
+                    if let Some(value) = value {
+                        let mut os = value.os();
+                        os.push(suffix.as_ref());
+                        Ok(Arg::Named {
+                            name: name.to_owned(),
+                            value: Some(OsOrStr::from(os)),
+                        })
+                    } else {
+                        Err(Error::fail("Unexpected trailing non-utf8 after a flag"))
```

Comment:
In this spec https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html item 2.b contains this:

> If the SYNOPSIS shows an optional option-argument (as with `[ -f[ option_argument]]` in the example), a conforming application shall place any option-argument for that option directly adjacent to the option in the same argument string, without intervening <blank> characters.

Now, if `-f` is valid utf8 (ascii), but `option_argument` starts with invalid utf8 split gives this tuple (`-f`, `option_argument`) so result is a named arg with the suffix as a value.

Reply:
Feels like we'd need to have `ascii_prefix_pos` and `split_str_param_pos` so we can potentially avoid reallocations and deal with this set of corner cases.